Optymalizacja

gwarser opened the pull request:
To powinno usunąć ślimaki w ABP.

Zamknięcie filtrów w `||...^` spowoduje, że będą pasować tylko do domen, a nie gdzieś w ścieżce.
Problemem może być to, że jeśli kiedyś ktoś dodał filtr bez gwiazdki dla domeny takiej jak `123oszustwo.pl` -> `oszustwo.pl` zamiast `*oszustwo.pl`, to teraz nie będą działały, bo "oszustwo" będzie traktowane jak domena a nie jej część. Druga sprawa, to ktoś mógł dodać filtr wyglądający jak domena (`oszustwo.pl`) ale w rzeczywistości będącego fragmentem ścieżki (`http://example.com/oszustwo.pl/asdf`), to teraz też nie będzie działał.

Dodanie gwiazdki do filtrów ścieżek (`/xxx/*`) spowoduje, że nie będą traktowane jako wyrażenia regularne tylko jako zwykłe ścieżki.
Tu chyba nie będzie problemów, bo we wszystkich filtrach były tylko zwykłe stringi.

radekg81 committed:
Usunięte duplikaty i zbędne spacje

radekg81 committed:
Zakotwicz filtry tylko do domeny

radekg81 committed:
Nie traktuj ścieżek jak regexy

azet12 commented:
Dzięki wielkie za zmiany i porady.
Mam jednak zastrzeżenie (albo czegoś nie rozumiem): reguły `/xxx/*` w ogóle mi nie działają na uBlocku. Np. `/apps.facebook.com/1880603628824303/*` po prostu przepuszcza - jak to niby ma działać?

gwarser commented:
Gdzie ona ma działać?

azet12 commented:
Ma po prostu blokować stronę `https://apps.facebook.com/1880603628824303`

gwarser commented:
Dla adresu `https://apps.facebook.com/1880603628824303` to mi żadna nie działa.

Żeby blokować w UBO całą stronę za pomocą strict-blocking to filtr musi być domeną albo mieć dodaną opcję `$document` (i subdocument dla ramek). W dodatku dałeś link bez `/` na końcu więc tu też nie pasuje teraz. Może gdy wykrywało filtr jako regex to miało to jakoś działać, ale testuję i  też nie działa ani w ABP ani w UBO.

Dla UBO chyba najlepiej by było tak:

`||apps.facebook.com/1880603628824303$document,subdocument`

ABP wcale nie blokuje całych stron, więc jak to działało w nim wcześniej to nie wiem jak to napisać.

gwarser commented:
Te też mają blokować strony?

```
/fejs-podgladacz/
/face-podgladacz/
/facebook-colors/
/face-colors/
/face-viewer/
/pakiety-pokemongo/
```

azet12 commented:
W sumie
```
/fejs-podgladacz/
/face-podgladacz/
/facebook-colors/
/face-colors/
/face-viewer/
/pakiety-pokemongo/
```
można całkowicie usunąć, dodałem te reguły do testów i jakoś zapomniałem, że to w ogóle istnieje.

ABP nie blokuje i nie blokował stron żadną regułą, dlatego uBlock jest najlepszy do KAD.

radekg81 committed:
Usunięte nieużywany filtry '/face...' itp.

azet12 commented:
Jest prawie idealnie, ale...
Przykład: `||bit.ly/2bJNp1u$document,$subdocument` - `$subdocument` powoduje, że reguła w ogóle nie działa - strona jest przepuszczana.

gwarser commented:
I zastanawiam się czy nie lepiej by było użyć `$document,~document` (blokuj dokument i niedokument), to by wtedy blokowało też inne zasoby z tych stron. Nie wiem czy to najlepsze rozwiązanie - gdzieś to już widziałem, ale nie mogę znaleźć.

azet12 commented:
Nie mam pojęcia skąd ja tam wziąłem ten `$`, ja chyba jakiś niepoważny dzisiaj jestem. Dobra mniejsza z tym. Wszystko działa.

Inne zasoby stron, czyli co? Strony w tej sekcji to są strony typu bitly - nie robią nic, oprócz przekierowania.

gwarser commented:
Ale chyba można też przez nie dać linka do jakiegoś skryptu/obrazka?

azet12 commented:
Na stronach z przekrętami czasami się spotykam iframe z bitly (lub podobnej strony), ale to tyle. Oszuści chcą po prostu wiedzieć ile mieli clicków, dużo sztuczek z tym bitly to się nie da zrobić.

```
||gold-lead.pl/pusta_oferta/api.php
||gold-lead.pl/vid_lock/video.php
||www.adworkmedia.com/gTemplate.php
```
to są linki do fałszywych widżetów, gdzie trzeba wpisać swój numer telefonu - zwykle pojawiają się w iframe. Ale teraz sprawdziłem, te dwa pierwsze są offline.

azet12 committed:
Update KAD.txt

Dostosowanie filtrów eksperymentalnych i obrazkowych. Aktualizacja wersji i daty.
